packages/story-editor/src/components/panels/design/textAccessibility/textAccessibility.js

```diff
@@ -0,0 +1,167 @@
+  const [currentTags, setCurrentTags] = useState([]);
+  useEffect(() => {
+    // We only want to impact elements if all selected elements are of type text
+    const textElements = selectedElements.filter(({ type }) => 'text' === type);
+    // Then we want to get the text tags for the elements if not already defined
+    const textTags = getTextElementTagNames(textElements);
+    // Then combine the elements with their associated tag
+    const newElements = combineElementsWithTags(textElements, textTags);
+    // And then set them into component state
+    setSelectedTextElements(newElements);
+    setCurrentTags(Array.from(textTags.values()));
+  }, [selectedElements, setSelectedTextElements]);
+  const { highlight, resetHighlight, cancelHighlight } = useHighlights(
+    (state) => ({
+      highlight: state[states.ASSISTIVE_TEXT],
+      resetHighlight: state.onFocusOut,
+      cancelHighlight: state.cancelEffect,
+    })
+  );
+  if (!showSemanticHeadings) {
+    return null;
+  }
+  // Map all types of tag names in the selected elements
+  // and then convert to an Array for usage
+  const handleChange = (ev, value) => {
+    selectedElements.map(() => {
+      return pushUpdate({ tagName: value });
+    });
+    // Update the selected text elements with the
+    // value chosen in the text accessibility dropdown
+    const newTags = getTextElementTagNames(selectedTextElements, value);
+    setCurrentTags(Array.from(newTags.values()));
+  };
+  // Check if tags match for selected item in dropdown
+  // If they don't match, we want to show 'Mixed'
+  // However, if they match, show the matching value
+  const currentValue =
+    new Set(currentTags || []).size === 1 ? currentTags[0] : '((MULTIPLE))';
+  return (
+    <SimplePanel
+      css={highlight && styles.FLASH}
+      onAnimationEnd={() => resetHighlight()}
+      name="textAccessibility"
+      title={__('Accessibility', 'web-stories')}
+      isPersistable={!highlight}
+    >
+      <Row>
+        <DropDown
+          ref={(node) => {
+            if (node && highlight?.focus && highlight?.showEffect) {
+              node.addEventListener('keydown', cancelHighlight, { once: true });
+              node.focus();
+            }
+          }}
+          data-testid="text-accessibility-dropdown"
+          title={__('Heading Levels', 'web-stories')}
+          dropdownButtonLabel={__('Heading Levels', 'web-stories')}
+          options={semanticHeadingsOptionsMap}
+          selectedValue={currentValue}
+          placeholder={
+            MULTIPLE_VALUE === currentValue
+              ? MULTIPLE_DISPLAY_VALUE
+              : currentValue
+          }
+          onMenuItemClick={handleChange}
+          dropDownLabel={__('Heading Level', 'web-stories')}
+        />
+      </Row>
+      <Text size={THEME_CONSTANTS.TYPOGRAPHY.PRESET_SIZES.X_SMALL}>
+        <TranslateWithMarkup
+          mapping={{
+            a: (
+              <a
```

Comment:
Added: https://github.com/GoogleForCreators/web-stories-wp/pull/11888/commits/0406524707b2ba47c41ac958eaccac4567c1d98d

Just made it dashes instead of underscores based on some other usage I saw in the application, but if there's a pattern for naming I'm missing and underscore is preferred, let me know.

Reply:
Ah, that totally makes sense. Can do, and will change the underscores, thanks.

Reply:
Addressed in https://github.com/GoogleForCreators/web-stories-wp/pull/11888/commits/705ece200af422cea010cd4b6b17fe5843cddee3